addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -218,6 +218,11 @@ const UserValueWidget = Widget.extend({
             params.activeValue = this.getActiveValue(methodName);
             params.defaultValue = this.getDefaultValue(methodName);
         }
+        const activeWidget = this._userValueWidgets.find(w => w.isActive());
+        if (activeWidget) {
+            const activeWidgetParams = activeWidget.getMethodsParams();
+            params.records = activeWidgetParams.records;
```

Comment:
@qsm-odoo Here I have to get the records param of the select's m2o child UserValueWidget.
This is I think the easiest way to do so but not really pretty...
Sould I do something more generic like append all of the activeWidget params to like params.activeWidgetParams
Or keep it really tight to m2o and verify that activeWidget is a m2o ?

Reply:
I don't understand. Your new widget does not have any child, it is just an input. You should not have to do anything here I guess

Reply:
It can have parent which is a select so the select will be the one handling the data-method-name and not the input. The select does not have records in it's params.

Reply:
Mmh, ok I guess but if it works to have anything else than a button inside a select, it is a miracle (I kinda write code for potentially have that but... never tested). Better to have `activeWidgetParams` then or simply merge them with the actual params (I think better). In any case: this should not be generic to UserValueWidget but the select one, having one active widget is the property of a select, not any UserValueWidget.

Reply:
By merging I might put the name of the active widget as the name of the select in the params if the select does not have any names. I think there is some potential bugs that can be avoided by having `activeWidgetParams`